src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedBrowserStorage.cs

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// Provides mechanisms for storing and retrieving data in the browser storage.
+    /// </summary>
+    public abstract class ProtectedBrowserStorage
+    {
+        private const string JsFunctionsPrefix = "Blazor._internal.protectedBrowserStorage";
+
+        private readonly string _storeName;
+        private readonly IJSRuntime _jsRuntime;
+        private readonly IDataProtectionProvider _dataProtectionProvider;
+        private readonly ConcurrentDictionary<string, IDataProtector> _cachedDataProtectorsByPurpose
+            = new ConcurrentDictionary<string, IDataProtector>();
+
+        // Stylistically, it doesn't matter at all what options we choose, since the values
+        // will be opaque after data protection. All that matters is that some fixed set of
+        // options exists and remains constant forever. We should choose whatever options
+        // maximize the ability to round-trip .NET objects reliably.
+        private readonly static JsonSerializerOptions SerializerOptions = new JsonSerializerOptions();
```

Comment:
This would make the serialized JSON properties case-insensitive. Would we want to support models such as:
```
class TestModel
{
    public string Id { get; set; }
    public string ID { get; set; }
}
```
?

Reply:
That's very unusual. Plus if someone absolutely needs to do something like this, they could pass a json serialized string. For now, it would be to fine to keep the same round-tripping behavior as IJSRuntime.